src/ring_theory/ideal/operations.lean

```diff
@@ -980,6 +988,31 @@ theorem map_inf_le : map f (I ⊓ J) ≤ map f I ⊓ map f J :=
 theorem le_comap_sup : comap f K ⊔ comap f L ≤ comap f (K ⊔ L) :=
 (gc_map_comap f).monotone_u.le_map_sup _ _
 
+@[simp] lemma smul_top_eq_map {R S : Type*} [comm_semiring R] [comm_semiring S] [algebra R S]
+  (I : ideal R) : I • (⊤ : submodule R S) = (I.map (algebra_map R S)).restrict_scalars R :=
```

Comment:
Is this also true as
```suggestion
  (I : ideal R) : I • (⊤ : submodule R S) = (submodule.map I (algebra.linear_map R S)) :=
```

Reply:
No, `submodule.map` is not closed under multiplication by `S` while `ideal.map` is. In particular if `f` is not surjective, `submodule.map f \top` is just the range of `f` while `ideal.map f \top` contains every element of `S`.

Reply:
(Which is why I also added `submodule.map_le_smul_top` above, and not `map_eq_smul_top`.)